Fix documentation about install and renew your own CA certificates (#6240)
strimzi/strimzi-kafka-operator: documentation/modules/security/proc-installing-your-own-ca-certificates.adoc

The Cluster Operator automatically generates and renews the following secrets:

`_CLUSTER-NAME_-cluster-ca`:: The cluster secret that contains the private key for the cluster CA.

> `_CLUSTER-NAME_-cluster-ca-cert`:: The cluster secret that contains a cluster CA certificate. The certificate contains a public key to validate the identity of Kafka brokers and ZooKeeper nodes.

Reviewer: This is user facing docs => users do not connect to ZooKeeper ... so I think this is confusing. I think it should remain as it was.

* Label `strimzi.io/kind=Kafka` identifies the Kafka custom resource.
* Label `strimzi.io/cluster=_CLUSTER-NAME_` identifies the Kafka cluster.

. Annotate the secrets
+
[source,shell,subs="+quotes"]
----
kubectl annotate secret _CA-CERTIFICATE-SECRET_ strimzi.io/ca-cert-generation=_CA-CERTIFICATE-GENERATION_
----
+
[source,shell,subs="+quotes"]
----
kubectl annotate secret _CA-KEY-SECRET_ strimzi.io/ca-key-generation=_CA-KEY-GENERATION_
----
+
* Annotation `strimzi.io/ca-cert-generation=_CA-CERTIFICATE-GENERATION_` defines the generation of a new CA certificate.
* Annotation `strimzi.io/ca-key-generation=_CA-KEY-GENERATION_` defines the generation of a new CA key.
+

> If you are replacing CA certificates automatically generated by the Cluster Operator, use the next higher incremental value from the existing annotation.

Reviewer: I'm confused. Does this refer to migration from Strimzi generated CA to custom CA?

Author: Yes. I thought it was important to highlight this if user starts with Strimzi CAs and then wants to move to his own CA certs. For triggering the update, increasing the generation is needed.

> If you are replacing CA certificates automatically generated by the Cluster Operator, use the next higher incremental value from the existing annotation.

Reviewer: Sorry to be picky. I think the idea to cover the migration from Strimzi generated to custom CA is good and useful. But I think it needs to point to the renewal guide and not suggest to follow the _installing your own CA_ guide. This guide goes through the creation of new secret before deploying cluster. You cannot do that for existing cluster with Strimzi's CA. You would also need to follow the _new key_ renewal pattern to ensure continuity. Just creating a brand new secret with new certs would break the cluster ragardless what generation it uses.

It would also need to include the changes to the Kafka configuration. So maybe it needs separate procedure?

Reviewer: I think this is still valid comment.

> If you are replacing CA certificates automatically generated by the Strimzi cluster operator then you should use the next incremental value as new generation from the corresponding existing annotation.

Author: @PaulRMellor didn't get what you mean sorry ... which procedure are you referring to?

> If you are replacing CA certificates automatically generated by the Strimzi cluster operator then you should use the next incremental value as new generation from the corresponding existing annotation.

Author: Isn't the user confused that he should set always generation 1? I mean if the current has generation 0 then 1 is ok ... but even 2, 3, 10, 50, ... any number greater than the current.

> If you are replacing CA certificates automatically generated by the Strimzi cluster operator then you should use the next incremental value as new generation from the corresponding existing annotation.

Reviewer 2: I added the example we use in the other step.

> If you are generating the CA certificates for the first time, you can start from 0 (zero) as the incremental value.

Reviewer 2: ```suggestion
If you are generating the CA certificates for the first time, you can start from 0 (zero) as the incremental value (`strimzi.io/ca-cert-generation=0`). This value must be incremented when you renew the certificates.

> If you are generating the CA certificates for the first time, you can start from 0 (zero) as the incremental value (`strimzi.io/ca-cert-generation=0`). Set a higher incremental value when you renew the certificates.

Reviewer: ```suggestion
If you are generating the CA certificates for the first time, you can start from 0 (zero) as the incremental value (for example `strimzi.io/ca-cert-generation=0`).
```

Reviewer: This as well.
